connectors/couchbase/connector-couchbase/src/main/java/org/teiid/resource/adapter/couchbase/CouchbaseManagedConnectionFactory.java

```diff
@@ -0,0 +1,309 @@
+        this.connectTimeout = connectTimeout;
+    }
+
+    public Boolean getDnsSrvEnabled() {
+        return dnsSrvEnabled;
+    }
+
+    public void setDnsSrvEnabled(Boolean dnsSrvEnabled) {
+        this.dnsSrvEnabled = dnsSrvEnabled;
+    }
+
+    public String getConnectionString() {
+        return connectionString;
+    }
+
+    public void setConnectionString(String connectionString) {
+        this.connectionString = connectionString;
+    }
+
+    public String getKeyspace() {
+        return keyspace;
+    }
+
+    public void setKeyspace(String keyspace) {
+        this.keyspace = keyspace;
+    }
+
+    public String getNamespace() {
+        return namespace;
+    }
+
+    public void setNamespace(String namespace) {
+        this.namespace = namespace;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public String getTimeUnit() {
+        return timeUnit;
+    }
+
+    public void setTimeUnit(String timeUnit) {
+        this.timeUnit = timeUnit;
+    }
+
+    @SuppressWarnings("serial")
+    @Override
+	public BasicConnectionFactory<CouchbaseConnectionImpl> createConnectionFactory() throws ResourceException {
+		
+		final CouchbaseEnvironment environment = DefaultCouchbaseEnvironment.builder()
+                .managementTimeout(managementTimeout)
+                .queryTimeout(queryTimeout)
+                .viewTimeout(viewTimeout)
+                .kvTimeout(kvTimeout)
+                .searchTimeout(searchTimeout)
+                .connectTimeout(connectTimeout)
+                .dnsSrvEnabled(dnsSrvEnabled)
+                .build();
+		
+		if (this.connectionString == null) {
+            throw new InvalidPropertyException(UTIL.getString("no_server")); //$NON-NLS-1$
+        }
+		
+		if (this.keyspace == null) {
+            throw new InvalidPropertyException(UTIL.getString("no_keyspace")); //$NON-NLS-1$
+        }
+		
+		if (this.namespace == null) {
+            throw new InvalidPropertyException(UTIL.getString("no_namespace")); //$NON-NLS-1$
+        }
+		
+		TimeUnit unit = TimeUnit.MILLISECONDS;
+		if(this.timeUnit != null) {
+		    try {
+                unit = TimeUnit.valueOf(timeUnit);
+            } catch (IllegalArgumentException e) {
+                throw new InvalidPropertyException(UTIL.getString("invalid_timeUnit", timeUnit)); //$NON-NLS-1$
+            }
+		}
+		final TimeUnit timeoutUnit = unit;
+
+		return new BasicConnectionFactory<CouchbaseConnectionImpl>(){
+
+            @Override
+            public CouchbaseConnectionImpl getConnection() throws ResourceException {
+                return new CouchbaseConnectionImpl(environment, connectionString, keyspace, password, timeoutUnit, namespace);
+            }};
+		
+	}
+
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + this.getManagementTimeout().hashCode();
```

Comment:
I do not think you want to include any timeout values to be part of object equality